app/src/main/java/fr/free/nrw/commons/upload/FileProcessor.kt

```diff
@@ -163,29 +173,44 @@ class FileProcessor @Inject constructor(
      *
      * @param imageCoordinates
      */
-    fun useImageCoords(imageCoordinates: ImageCoordinates) {
+    fun prePopulateCategoriesAndDepictionsBy(imageCoordinates: ImageCoordinates) {
         requireNotNull(imageCoordinates.decimalCoords)
-        cacheController.setQtPoint(imageCoordinates.decLongitude, imageCoordinates.decLatitude)
-        val displayCatList = cacheController.findCategory()
-
-        // If no categories found in cache, call MediaWiki API to match image coords with nearby Commons categories
-        if (displayCatList.isEmpty()) {
-            compositeDisposable.add(
-                apiCall.request(imageCoordinates.decimalCoords)
-                    .subscribeOn(Schedulers.io())
-                    .observeOn(Schedulers.io())
-                    .subscribe(
-                        { gpsCategoryModel.categoryList = it },
-                        {
-                            Timber.e(it)
-                            gpsCategoryModel.clear()
-                        }
-                    )
+        compositeDisposable.add(
+            apiCall.request(imageCoordinates.decimalCoords)
+                .subscribeOn(Schedulers.io())
+                .observeOn(Schedulers.io())
+                .subscribe(
+                    { gpsCategoryModel.categoryList = it },
+                    {
+                        Timber.e(it)
+                        gpsCategoryModel.clear()
+                    }
+                )
+        )
+
+        compositeDisposable.add(
+            suggestNearbyDepictions(imageCoordinates)
+        )
+    }
+
+    private val radiiProgressionInMetres =
+        (DEFAULT_SUGGESTION_RADIUS_IN_METRES..MAX_SUGGESTION_RADIUS_IN_METRES step RADIUS_STEP_SIZE_IN_METRES)
```

Comment:
Great! Actually whenever I need to search for items near a point, I always use linear radii progression, as it results in a square surface progression.